valhalla/mjolnir/landmark_builder.h

```diff
@@ -0,0 +1,164 @@
+      {{"restaurant", LandmarkType::restaurant},
+       {"school", LandmarkType::school},
+       {"fast_food", LandmarkType::fast_food},
+       {"cafe", LandmarkType::cafe},
+       {"fuel", LandmarkType::fuel},
+       {"bank", LandmarkType::bank},
+       {"pharmacy", LandmarkType::pharmacy},
+       {"kindergarten", LandmarkType::kindergarten},
+       {"bar", LandmarkType::bar},
+       {"hospital", LandmarkType::hospital},
+       {"post_office", LandmarkType::post_office},
+       {"pub", LandmarkType::pub},
+       {"clinic", LandmarkType::clinic},
+       {"police", LandmarkType::police},
+       {"fire_station", LandmarkType::fire_station},
+       {"car_wash", LandmarkType::car_wash},
+       {"charging_station", LandmarkType::charging_station},
+       {"college", LandmarkType::college},
+       {"university", LandmarkType::university},
+       {"theatre", LandmarkType::theatre},
+       {"cinema", LandmarkType::cinema}};
+
+  auto it = string_to_landmark_type.find(s);
+  if (it != string_to_landmark_type.cend()) {
+    return it->second;
+  }
+  return LandmarkType::null;
+}
+
+inline std::string landmark_type_to_string(LandmarkType t) {
+  static const std::unordered_map<uint8_t, std::string> landmark_type_to_string =
+      {{static_cast<uint8_t>(LandmarkType::null), "na"},
+       {static_cast<uint8_t>(LandmarkType::restaurant), "restaurant"},
+       {static_cast<uint8_t>(LandmarkType::school), "school"},
+       {static_cast<uint8_t>(LandmarkType::fast_food), "fast_food"},
+       {static_cast<uint8_t>(LandmarkType::cafe), "cafe"},
+       {static_cast<uint8_t>(LandmarkType::fuel), "fuel"},
+       {static_cast<uint8_t>(LandmarkType::bank), "bank"},
+       {static_cast<uint8_t>(LandmarkType::pharmacy), "pharmacy"},
+       {static_cast<uint8_t>(LandmarkType::kindergarten), "kindergarten"},
+       {static_cast<uint8_t>(LandmarkType::bar), "bar"},
+       {static_cast<uint8_t>(LandmarkType::hospital), "hospital"},
+       {static_cast<uint8_t>(LandmarkType::post_office), "post_office"},
+       {static_cast<uint8_t>(LandmarkType::pub), "pub"},
+       {static_cast<uint8_t>(LandmarkType::clinic), "clinic"},
+       {static_cast<uint8_t>(LandmarkType::police), "police"},
+       {static_cast<uint8_t>(LandmarkType::fire_station), "fire_station"},
+       {static_cast<uint8_t>(LandmarkType::car_wash), "car_wash"},
+       {static_cast<uint8_t>(LandmarkType::charging_station), "charging_station"},
+       {static_cast<uint8_t>(LandmarkType::college), "college"},
+       {static_cast<uint8_t>(LandmarkType::university), "university"},
+       {static_cast<uint8_t>(LandmarkType::theatre), "theatre"},
+       {static_cast<uint8_t>(LandmarkType::cinema), "cinema"}};
+
+  auto it = landmark_type_to_string.find(static_cast<uint8_t>(t));
+  if (it != landmark_type_to_string.cend()) {
+    return it->second;
+  }
+  return "null";
+}
+
+struct Landmark {
+  std::string name;
+  LandmarkType type;
```

Comment:
the way we're currently using this struct, we should default at least the name & type or you'll get random stuff in there if they're never set.

Reply:
ah right I forgot default / initialization again. let's set default name as `"null"` and default type as `LandmarkType::null` :D

Reply:
i disagree here. i think we shouldnt even h ave this struct but rather for the insert we can take the parameters in the function (why make a copy when we dont need to) and then when returning we can just return a tuple. as far as defaults, name can either exist or be empty we dont need a name for all amenities the type is good enough for the majority. default location makes no sense, we should never have a landmark that doesnt have a location. again i think we can avoid this by simply removing this structure completely from the interface

Reply:
I can remove the struct. it simply makes the code prettier and easier so no problem. As for default value of name I think having a default `"null" `or `""` or none at all are interchangeable to me. which one is better from your opinion?
but I don't know how to check validity of lat and lng. other implementation files also don't deal with that so I assume it is either checked in parser or just doesn't exist, or both. so for `lat` and `lng` maybe we can keep the current with no defaults. WDYT? @nilsnolde

Reply:
in the database, it can be NULL but in code it has to be the empty string. therefor to make our lives easier it could also be empty string in the database